cmd/opampsupervisor/supervisor/commander/commander.go

```diff
@@ -52,6 +53,13 @@ func (c *Commander) Start(ctx context.Context) error {
 		return nil
 	}
 
+	if c.doneCh != nil {
+		close(c.doneCh)
+	}
```

Comment:
```suggestion
	if len(c.doneCh) != 0 {
		<-c.doneCh
	}
```

Closing the channel will fire a message on `doneCh`. I think we should create the channel in `NewCommander`, then drain it if it isn't empty using the snippet I proposed. This also means we don't change the channel instance, which I think has benefits of its own.